### PR TITLE
Add alternate attack procs

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds alternate attack procs for right-clicking. Updates the gunpoint code to use this new proc as an example (though I suspect this is being changed soon, so oops).

## Why It's Good For The Game
Makes it easier to add new behavior to right click, which we should be doing.

### CODE REVIEW

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -55,6 +81,19 @@
...
+ *
+ * Arguments:
+ * * obj/item/weapon - The item hitting this atom
+ * * mob/user - The wielder of this item
+ * * params - click params such as alt/shift etc
+ *
+ * See: [/obj/item/proc/melee_attack_chain]
+ */
+/atom/proc/attackby_alt(obj/item/weapon, mob/user, params)
+	return ALT_ATTACK_CALL_NORMAL
```
Comment (Jared-Fogle) [inline code/_onclick/item_attack.dm:95]: I don't really like this as the default, but there's a toooon of places where right clicking exists now, and I sort of just didn't want to go through all of it. This is for backwards compatibility.

Comment (Jared-Fogle) [reply code/_onclick/item_attack.dm:95]: Actually fuck it, I'm doing it.

Comment (Jared-Fogle) [reply code/_onclick/item_attack.dm:95]: Actually, on second thought, this makes more sense than I first thought, a lot of alternate attack code is being used in the middle of attackby's for seemingly good reason.

**Comment by Qustinnus** [comment]
port the existing procs to this if youre gonna add  this

**Comment by Jared-Fogle** [comment]
I still don't know if it's a good idea, I keep going back and forth on it.

For example, the classic baton's alternate check is waaaaaay underneath a ton of other code, and so it gives me pause on whether or not it's a good idea to move all of that out. I may, I may not.

**Comment by oklews** [comment]
This is 100% the justification I needed to be happy with new combat. This is going to be the next big thing.

**Comment by necromanceranne** [comment]
Nice was waiting a bit on this

**Pull request merged**